openedx/openedx-platform
# Replace Paver quality and js_test commands [Paver Removal 2/3]

Paver is deprecated, and we aim to fully remove it soon. You can find more details here:

[Paver deprecation issue](https://github.com/openedx/edx-platform/issues/34467)
[Paver removal PR](https://github.com/openedx/edx-platform/pull/34832)

A few Paver commands remain in the edx-platform, and to proceed with its removal, we've replaced the following Paver commands with equivalent Make commands:

1. Replaced paver run_pep8 with pycodestyle
2. Replaced paver run_eslint
3. Replaced paver run_stylelint
4. Replaced paver run_xsslint
5. Replaced paver run_pii_check
6. Replaced paver check_keywords
7. Replaced paver test_js* and paver diff_coverage

How to Test This PR:

In the edx-platform directory, you can use the following terminal commands to check for linting issues in Python and JavaScript files. Make sure your virtual environment is activated before running these commands.

`make pycodestyle`
`make eslint`
`make stylelint`
`make xsslint`
`make pi_check`
`make check_keyword`
`make test-js`
`make test-coverage`
`make quality-test`

For more details, please refer to the ticket: [Replace paver quality and js_test commands](https://github.com/orgs/openedx/projects/55/views/1?pane=issue&itemId=63954762)
